provide a way to avoid re2 installation

Unless I am missing something, `re2` is added as a dependency but it is not actually being used anywhere.

+1

just started having a related issue building a backend service using metascraper

```
#!/bin/bash -eo pipefail
docker build --rm=false -t gcr.io/${GCP_PROJECT}/${SERVICE_NAME}:$CIRCLE_TAG .
Sending build context to Docker daemon  13.47MB
Step 1/8 : FROM node:18.17.0-alpine
18.17.0-alpine: Pulling from library/node

a8db6415: Pulling fs layer 
2f1a5d31: Pulling fs layer 
b7606c1a: Pulling fs layer 
Digest: sha256:58878e9e1ed3911bdd675d576331ed8838fc851607aed3bb91e25dfaffab3267
Status: Downloaded newer image for node:18.17.0-alpine
 ---> f1fac320ae0c
Step 2/8 : WORKDIR /home/app
 ---> Running in 048cc5001d44
 ---> 8fcfa90cc1b8
Step 3/8 : COPY package.json package-lock.json /home/app/
 ---> d8292bd4b24a
Step 4/8 : RUN npm config set update-notifier false
 ---> Running in 709aca441111
 ---> 0ffe21789570
Step 5/8 : RUN npm install --quiet --production
 ---> Running in 81143fdafca6
npm WARN config production Use `--omit=dev` instead.
npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
npm ERR! code 1
npm ERR! path /home/app/node_modules/re2
npm ERR! command failed
npm ERR! command sh -c install-from-cache --artifact build/Release/re2.node --host-var RE2_DOWNLOAD_MIRROR --skip-path-var RE2_DOWNLOAD_SKIP_PATH --skip-ver-var RE2_DOWNLOAD_SKIP_VER || npm run rebuild
npm ERR! Trying https://github.com/uhop/node-re2/releases/download/1.20.1/linux-musl-x64-108.br ...
npm ERR! Writing to build/Release/re2.node ...
npm ERR! The verification has failed: building from sources ...
npm ERR! Building locally ...
npm ERR! 
npm ERR! > re2@1.20.1 rebuild
npm ERR! > node-gyp rebuild
npm ERR! 
npm ERR! 
npm ERR! > re2@1.20.1 rebuild
npm ERR! > node-gyp rebuild
npm ERR! gyp ERR! find Python 
npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
```

it seems re2 is failing and it is trying to build locally (the build worked before).

--- a/README.md
+++ b/README.md
@@ -279,6 +279,15 @@ Default: `true`
 
 Ensure the URL provided is validated as a [WHATWG URL](https://nodejs.org/api/url.html#url_the_whatwg_url_api) API compliant.
 
+## Environment Variables
+
+#### METASCRAPER_RE2
+
+Type: `boolean`<br>
+Default: `true`
+
+It attemptt to load re2 to use instead of RegExp.
+
 ## Benchmark
 
 To give you an idea of how accurate **metascraper** is, here is a comparison of similar libraries:

--- a/packages/metascraper-helpers/index.js
+++ b/packages/metascraper-helpers/index.js
@@ -1,7 +1,6 @@
 'use strict'
 
 const memoizeOne = require('memoize-one').default || require('memoize-one')
-const urlRegex = require('url-regex-safe')({ exact: true, parens: true })
 const condenseWhitespace = require('condense-whitespace')
 const { getExtension: mimeExtension } = require('mime')
 const capitalize = require('microsoft-capitalize')
@@ -18,6 +17,14 @@ const isIso = require('isostring')
 const isUri = require('is-uri')
 const { URL } = require('url')
 const tldts = require('tldts')
+
+const urlRegex = require('url-regex-safe')({
+  exact: true,
+  parens: true,
+  re2: process.env.METASCRAPER_RE2
+    ? process.env.METASCRAPER_RE2 === 'true'
+    : undefined
+})
 
 const {
   chain,

--- a/packages/metascraper-helpers/package.json
+++ b/packages/metascraper-helpers/package.json
@@ -42,7 +42,7 @@
     "re2": "~1.20.1",
     "smartquotes": "~2.3.2",
     "tldts": "~6.0.12",
-    "url-regex-safe": "~3.0.0",
+    "url-regex-safe": "~4.0.0",
     "video-extensions": "~1.2.0"
   },
   "devDependencies": {

